react-app/src/components/ChartsIndividual.js

```diff
@@ -25,74 +26,99 @@ const useStyles = makeStyles((theme) => ({
 }));
 
 
-async function getAllCdns(){
+async function getAllCdns() {
   return fetch(process.env.REACT_APP_API_URL + "/api/parameters")
     .then(response => response.json())
     .then(data => data['cdns'].map(cdn => cdn.name))
-    .then(d =>{
+    .then(d => {
       console.log(d);
       return d;
     })
 }
 
+async function getAllAvailableMonitors() {
+  return fetch(process.env.REACT_APP_API_URL + "/api/monitors")
+    .then(response => response.json())
+    .then(data => data['monitors'].map(monitor => monitor.address))
+    .then(a => {
+      const choices = ["LOCAL", "ALL", "COMPARE"];
+      a.map(m => choices.push(m));
+      console.log(choices);
+      return choices;
+    })
+}
 
-const ChartsIndividual = (cdns) => {
+const ChartsIndividual = (monitors, cdns) => {
   const classes = useStyles();
   const [cdn, setCDN] = useState("");
   const [allCdnsItems, setAllCdnsItems] = useState([]);
-
-
-
+  const [monitor, setMonitor] = useState("");
+  const [allMonitorsItems, setAllMonitorsItems] = useState([]);
 
   useEffect(() => {
-    getAllCdns().then( cdns => {
-      let items = cdns.map(c => {
-        return <MenuItem value={c}>{c}</MenuItem>
-      });
-      setAllCdnsItems(items);
-      setCDN(cdns[0]);// TODO handle no cdns
-
-    }).catch(error => console.log(error))
-  }, []
+      getAllCdns().then(cdns => {
+        let items = cdns.map(c => {
+          return <MenuItem value={c}>{c}</MenuItem>
+        });
+        setAllCdnsItems(items);
+        setCDN(cdns[0]);// TODO handle no cdns
+      }).catch(error => console.log(error));
+      getAllAvailableMonitors().then(monitors => {
+        let items = monitors.map(m => {
+          return <MenuItem value={m}>{m}</MenuItem>
+        });
+        setAllMonitorsItems(items);
+        setMonitor(monitors[0]);// TODO handle no monitors
+      }).catch(error => console.log(error))
+    }, []
   )
   const handleChange = (event) => {
     setCDN(event.target.value);
+    setMonitor(event.target.value);
```

Comment:
Shouldn't we use separate callbacks for cdn and monitor instead of one handleChange?